## US Birth Data Set

[Original article by FiveThirtyEight about Friday the 13th](http://fivethirtyeight.com/features/some-people-are-too-superstitious-to-have-a-baby-on-friday-the-13th/)

The data set contains U.S. births data for the years 1994 to 2003, as provided by the Centers for Disease Control and Prevention's National Center for Health Statistics

### Assignment

- Open the CSV and split based on new lines
- Preview the first 10 entries

In [1]:
import pandas
birth_data = pandas.read_csv('us_births.csv', nrows=10)

### Assignment

- Create a function that takes in a CSV and converts the data into a list of lists
    - Each row will be a list
    - Make sure to convert the values to int
    - Return the final list of lists
- Preview the first 10 entries of the output

In [2]:
def convert_csv_to_list(csv_file):
    result_list = []
    birth_data = pandas.read_csv(csv_file)
    for index, row in birth_data.iterrows():
        row_list = [row['year'], row['month'], row['date_of_month'], row['day_of_week'], row['births']]
        #print row_list
        result_list.append(row_list)
    return result_list

us_birth_list = convert_csv_to_list('us_births.csv')
print us_birth_list[0:10]

[[1994, 1, 1, 6, 8096], [1994, 1, 2, 7, 7772], [1994, 1, 3, 1, 10142], [1994, 1, 4, 2, 11248], [1994, 1, 5, 3, 11053], [1994, 1, 6, 4, 11406], [1994, 1, 7, 5, 11251], [1994, 1, 8, 6, 8653], [1994, 1, 9, 7, 7910], [1994, 1, 10, 1, 10498]]


### Assignment

- Create a function that calculates the number of births each month
    - The function input should be the previous list of lists you created
    - Use a dictionary and increment the values associated with each month key
    - Return the final dictionary
- Preview the output

[1994, 1, 1, 6, 8096]
[1994, 1, 2, 7, 7772]
[1994, 1, 3, 1, 10142]
[1994, 1, 4, 2, 11248]
[1994, 1, 5, 3, 11053]
[1994, 1, 6, 4, 11406]
[1994, 1, 7, 5, 11251]
[1994, 1, 8, 6, 8653]
[1994, 1, 9, 7, 7910]
[1994, 1, 10, 1, 10498]

In [3]:
def calculate_num_of_births_by_month(input_list):
    result_dict = {}
    
    # initialize dictionary with default values
    for month in range(1,13):
        result_dict[month] = 0
    
    #print result_dict
    for row in input_list:
        result_dict[row[1]] += row[4] 

    return result_dict
calculate_num_of_births_by_month(us_birth_list)

{1: 3232517,
 2: 3018140,
 3: 3322069,
 4: 3185314,
 5: 3350907,
 6: 3296530,
 7: 3498783,
 8: 3525858,
 9: 3439698,
 10: 3378814,
 11: 3171647,
 12: 3301860}

### Assignment

- Create a function that calculates the number of births each day of the week

In [4]:
def calculate_num_of_births_by_day(input_list):
    result_dict = {}
    
    # initialize dictionary with default values
    for month in range(1,8):
        result_dict[month] = 0
    
    #print result_dict
    for row in input_list:
        result_dict[row[3]] += row[4]

    return result_dict
calculate_num_of_births_by_day(us_birth_list)

{1: 5789166,
 2: 6446196,
 3: 6322855,
 4: 6288429,
 5: 6233657,
 6: 4562111,
 7: 4079723}

### Assignment

- Create a general function that takes the data list of lists and a column index, it should return a dictionary mapping of the unique keys and summed values

In [5]:
def calculate_num_of_births_by_day(input_list, column_index):
    result_dict = {}
    unique_set_list = []

    # get a list of unique values by the column_index
    for row in input_list:
        unique_set_list.append(row[column_index])
        
    # initialize dictionary with default values
    for value in unique_set_list:
        result_dict[value] = 0
    
    # loop again to get the aggregate sum of values
    for row in input_list:
        result_dict[row[column_index]] += 1 
        
    return result_dict

# get the values by day of month
print calculate_num_of_births_by_day(us_birth_list, 2)

# get the values by day of week
print calculate_num_of_births_by_day(us_birth_list, 3)

{1: 120, 2: 120, 3: 120, 4: 120, 5: 120, 6: 120, 7: 120, 8: 120, 9: 120, 10: 120, 11: 120, 12: 120, 13: 120, 14: 120, 15: 120, 16: 120, 17: 120, 18: 120, 19: 120, 20: 120, 21: 120, 22: 120, 23: 120, 24: 120, 25: 120, 26: 120, 27: 120, 28: 120, 29: 112, 30: 110, 31: 70}
{1: 522, 2: 522, 3: 522, 4: 521, 5: 521, 6: 522, 7: 522}
